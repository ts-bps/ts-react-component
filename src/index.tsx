import * as React from "react";

export const sum = (a: number, b: number) => a + b;

export type SumProps = { a: number; b: number };

// export const Sum: React.StatelessComponent<SumProps> = ({
//   a,
//   b
// }) => {
//   return <>{sum(a, b)}</>;
// };

export class Sum extends React.Component<SumProps, {}> {
  s = async () => {
    return 2;
  };
  static defaultProps = {
    a: 0,
    b: 0
  };
  d = 2;
  async componentDidMount() {
    await this.s();
  }

  render() {
    const { a, b } = this.props;
    return <>{sum(a, b)}</>;
  }
}
