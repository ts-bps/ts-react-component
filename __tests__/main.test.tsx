import * as React from "react";
import * as Renderer from "react-test-renderer";
import { Sum } from "../src/main";

describe("Sum", () => {
  test("Renders", () => {
    const tree = Renderer.create(<Sum a={1} b={2} />);
    expect(tree).toMatchSnapshot();
  });
});
