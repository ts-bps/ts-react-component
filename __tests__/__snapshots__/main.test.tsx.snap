// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sum Renders 1`] = `"3"`;
